orchestrator/do-release/Jenkinsfile

```diff
@@ -92,13 +92,41 @@ node('docker') {
   def jenkinsAgentRootPath = "${JENKINS_AGENT_ROOT_PATH}"
   def exoUser = "${BUILD_USER_ID}"
   echo "* Projects: ${PROJECTS}"
+  echo "* Projects computed: ${projectsToRelease}"
   echo "* Command: ${RELEASE_CMD}"
   echo "* Projects: ${PROJECTS}"
   echo "* Releases in Parallel? ${isInParallel}"
 
+  stage "${JIRA_ID}-projects_validation"
+  def releasedProjects = [:]
+  def callback = { project -> 
+    def name = project.name
+    releasedProjects[name] = releasedProjects.containsKey(name) ? releasedProjects[name]+1 : 1
+  }
+  
+  doReleases(exoUser, jenkinsAgentRootPath, jiraID, projectsToRelease, releaseCMD, isInParallel, callback)
+
+  // check all the projects were found on the json
+  def unreleasedProjects = projectsToRelease - releasedProjects.keySet()
+  if(unreleasedProjects.length > 0) {
+    echo "ERROR The project(s) ${unreleasedProjects} were not found in the catalog file."
+    currentBuild.result = "FAIL"
+    return
+  }
+  // Check no project was present several times on the json
+  def duplicates = releasedProjects.findAll({ it.value > 1 }).keySet()
+  if(duplicates) {
+    echo "ERROR The project(s) ${duplicates} are released several times or are present several times on the json file."
+    currentBuild.result = "FAIL"
+    return
+  }
```

Comment:
I would move all the code related to validation in a method validateRelease (or something like that), it would make the code of the main method easy to ready and understand (validate, then release). WDYT ?

Reply:
done, for sure it will be cleaner.